test/router-options.test.js

```diff
@@ -58,3 +63,32 @@ test('Should honor maxParamLength option', t => {
     t.strictEqual(res.statusCode, 404)
   })
 })
+
+test('Should honor frameworkErrors option', t => {
+  t.plan(3)
+  const fastify = Fastify({
+    frameworkErrors: function (err, req, res) {
+      if (err instanceof FST_ERR_BAD_URL) {
+        t.ok(true)
+      } else {
+        t.fail()
+      }
+      res.end(err.message)
+    }
+  })
+
+  fastify.get('/test/:id', (req, res) => {
+    res.end('{ hello: \'world\' }')
+  })
+
+  fastify.inject(
+    {
+      method: 'GET',
+      url: '/test/%world'
+    },
+    (err, res) => {
+      t.error(err)
+      t.deepEquals(res.body, 'FST_ERR_BAD_URL: \'%world\' is not a valid url component')
```

Comment:
```suggestion
      t.equals(res.body, 'FST_ERR_BAD_URL: \'%world\' is not a valid url component')
```

`deepEquals` is for object